ArduCopter/Parameters.cpp

```diff
@@ -1413,103 +1370,26 @@ void Copter::load_parameters(void)
         // PARAMETER_CONVERSION - Added: Aug-2021
         { Parameters::k_param_pi_vel_xy, 3, AP_PARAM_FLOAT, "PSC_VELXY_FLTE" },
-        // PARAMETER_CONVERSION - Added: Jan-2018
-        { Parameters::k_param_p_vel_z, 0, AP_PARAM_FLOAT, "PSC_VELZ_P" },
-        { Parameters::k_param_pid_accel_z, 0, AP_PARAM_FLOAT, "PSC_ACCZ_P" },
-        { Parameters::k_param_pid_accel_z, 1, AP_PARAM_FLOAT, "PSC_ACCZ_I" },
-        { Parameters::k_param_pid_accel_z, 2, AP_PARAM_FLOAT, "PSC_ACCZ_D" },
-        { Parameters::k_param_pid_accel_z, 5, AP_PARAM_FLOAT, "PSC_ACCZ_IMAX" },
-        // PARAMETER_CONVERSION - Added: Oct-2019
-        { Parameters::k_param_pid_accel_z, 6, AP_PARAM_FLOAT, "PSC_ACCZ_FLTE" },
-        // PARAMETER_CONVERSION - Added: Jan-2018
-        { Parameters::k_param_p_alt_hold, 0, AP_PARAM_FLOAT, "PSC_POSZ_P" },
-        { Parameters::k_param_p_pos_xy, 0, AP_PARAM_FLOAT, "PSC_POSXY_P" },
-    };
-    const AP_Param::ConversionInfo loiter_conversion_info[] = {
-        // PARAMETER_CONVERSION - Added: Apr-2018
-        { Parameters::k_param_wp_nav, 4, AP_PARAM_FLOAT, "LOIT_SPEED" },
-        { Parameters::k_param_wp_nav, 7, AP_PARAM_FLOAT, "LOIT_BRK_JERK" },
-        { Parameters::k_param_wp_nav, 8, AP_PARAM_FLOAT, "LOIT_ACC_MAX" },
-        { Parameters::k_param_wp_nav, 9, AP_PARAM_FLOAT, "LOIT_BRK_ACCEL" }
     };
 
     // convert angle controller gain and filter without scaling
     for (const auto &info : angle_and_filt_conversion_info) {
         AP_Param::convert_old_parameter(&info, 1.0f);
     }
-    // convert RC_FEEL_RP to ATC_INPUT_TC
-    // PARAMETER_CONVERSION - Added: Mar-2018
-    const AP_Param::ConversionInfo rc_feel_rp_conversion_info = { Parameters::k_param_rc_feel_rp, 0, AP_PARAM_INT8, "ATC_INPUT_TC" };
-    AP_Int8 rc_feel_rp_old;
-    if (AP_Param::find_old_parameter(&rc_feel_rp_conversion_info, &rc_feel_rp_old)) {
-        AP_Param::set_default_by_name(rc_feel_rp_conversion_info.new_name, (1.0f / (2.0f + rc_feel_rp_old.get() * 0.1f)));
-    }
-    // convert loiter parameters
-    for (const auto &info : loiter_conversion_info) {
-        AP_Param::convert_old_parameter(&info, 1.0f);
-    }
 
     // TradHeli default parameters
 #if FRAME_CONFIG == HELI_FRAME
     static const struct AP_Param::defaults_table_struct heli_defaults_table[] = {
-        // PARAMETER_CONVERSION - Added: Nov-2018
-        { "LOIT_ACC_MAX", 500.0f },
-        { "LOIT_BRK_ACCEL", 125.0f },
-        { "LOIT_BRK_DELAY", 1.0f },
-        { "LOIT_BRK_JERK", 250.0f },
-        { "LOIT_SPEED", 3000.0f },
-        { "PHLD_BRAKE_ANGLE", 800.0f },
-        { "PHLD_BRAKE_RATE", 4.0f },
-        { "PSC_ACCZ_P", 0.28f },
-        { "PSC_VELXY_D", 0.0f },
-        { "PSC_VELXY_I", 0.5f },
-        { "PSC_VELXY_P", 1.0f },
-        // PARAMETER_CONVERSION - Added: Jan-2019
-        { "RC8_OPTION", 32 },
-        // PARAMETER_CONVERSION - Added: Aug-2018
+        // PARAMETER_CONVERSION - Added: Aug-2020
         { "RC_OPTIONS", 0 },
```

Comment:
optionally we could probably drop this RC_OPTIONS linte too which is from Aug 2020

Reply:
> optionally we could probably drop this RC_OPTIONS line too which is from Aug 2020

Yeah, I  really did consider it.  But they I couldn't use the pretty straight-forward 3.6.9 -> 4.x.x argument as to why it was time for these to go away :-)

I was also a little surprised at how much of the conversion code did go away.  I wonder if we're doing less of it now-adays...